liberty-maven-plugin/src/main/java/io/openliberty/tools/maven/server/GenerateFeaturesMojo.java

```diff
@@ -225,6 +230,26 @@ private void generateFeatures() throws MojoExecutionException, PluginExecutionEx
             }
         } catch (BinaryScannerUtil.FeatureUnavailableException featureUnavailable) {
             throw new MojoExecutionException(String.format(BinaryScannerUtil.BINARY_SCANNER_CONFLICT_MESSAGE5, featureUnavailable.getConflicts(), featureUnavailable.getMPLevel(), featureUnavailable.getEELevel(), featureUnavailable.getUnavailableFeatures()));
+        } catch (BinaryScannerUtil.IllegalTargetException illegalTargets) {
+            String messages = null;
+            if (illegalTargets.getEELevel() != null) {
+                messages = String.format(BinaryScannerUtil.BINARY_SCANNER_INVALID_EE_MESSAGE, eeVersion);
+            }
+            if (illegalTargets.getMPLevel() != null) {
+                if (messages != null) {
+                    messages += "\n" ;
+                    messages += String.format(BinaryScannerUtil.BINARY_SCANNER_INVALID_MP_MESSAGE, mpVersion);
+
+                } else {
+                    messages = String.format(BinaryScannerUtil.BINARY_SCANNER_INVALID_MP_MESSAGE, mpVersion);
+                }
+            }
+            if (messages == null) { // We need to be prepared for this situation from the binary scanner.
+                messages = BinaryScannerUtil.BINARY_SCANNER_INVALID_EEMPARG_MESSAGE;
+            }
```

Comment:
Any benefit in putting this in `BinaryScannerUtil` as a helper method since it looks similar across Maven/Gradle? Or is it such a small piece of code it may not be worth the effort/makes sense to duplicate

Reply:
I like the idea of a common helper method for this piece.

Reply:
I wasn't expecting it to be the same in LMP and LGP but since it is we can make a common method, thank you.